xdsl/dialects/onnx.py

```diff
@@ -256,6 +256,98 @@ def verify_(self) -> None:
             )
 
 
+@irdl_op_definition
+class Reshape(IRDLOperation):
+    """
+    Reshape the input tensor similar to numpy.reshape.
+    First input is the data tensor, second input is a shape tensor which specifies the output shape. It outputs the reshaped tensor.
+    At most one dimension of the new shape can be -1. In this case, the value is
+    inferred from the size of the tensor and the remaining dimensions. A dimension
+    could also be 0, in which case the actual dimension value is unchanged (i.e. taken
+    from the input tensor). Shape (second input) could be an empty shape, which means converting to a scalar.
+    The input tensor's shape and the output tensor's shape are required to have the same number of elements.
+    """
+
+    name = "onnx.Reshape"
+    T = Annotated[AnyFloat | IntegerType, ConstraintVar("T")]
+    data = operand_def(TensorType[T])
+    shape = operand_def(TensorType)
+    reshaped = result_def(TensorType[T])
+
+    allow_zero = opt_attr_def(IntegerAttr, attr_name="allowzero")
+
+    assembly_format = "`(` $data `,` $shape `)` attr-dict `:` `(` type($data) `,` type($shape) `)` `->` type($reshaped)"
+
+    def __init__(self, data: SSAValue, shape: SSAValue, allow_zero: Attribute):
+        super().__init__(
+            attributes={"allowzero": allow_zero},
+            operands=[data, shape],
+            result_types=[data.type],
+        )
+
+    def verify_(self) -> None:
+        if (
+            not isinstance(data_type := self.data.type, TensorType)
+            or not isinstance(shape_type := self.shape.type, TensorType)
+            or not isinstance(reshaped_type := self.reshaped.type, TensorType)
+        ):
+            assert (
+                False
+            ), "onnx elementwise operation operands and result must be of type TensorType"
+
+            data_type = cast(TensorType[Attribute], data_type)
+            reshaped_type = cast(TensorType[Attribute], reshaped_type)
+            shape_type = cast(TensorType[Attribute], shape_type)
+
+            if data_type != reshaped_type:
+                raise VerifyException("Mismatch between operand type and res type")
+            if shape_type.element_type != IntegerType(64):
```

Comment:
this is not exercised by a test case, can you please add one?

Moreover, on the same note, can you please check the code coverage report and increase test coverage as much as possible?